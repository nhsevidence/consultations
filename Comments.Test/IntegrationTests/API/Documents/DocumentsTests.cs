using Comments.Test.Infrastructure;
using Shouldly;
using System;
using System.Net;
using System.Threading.Tasks;
using NICE.Feeds.Tests.Infrastructure;
using Xunit;
using TestBase = Comments.Test.Infrastructure.TestBase;

namespace Comments.Test.IntegrationTests.API.Documents
{
    public class DocumentsTests : TestBase
    {
        [Theory]
        [InlineData(0)]
        [InlineData(-1)]
        public async Task Get_Documents_Feed_Using_Invalid_Consultation_Id_Throws_error(int consultationId)
        {
	        //Arrange (in the base constructor for this one.)

	        //Act
	        var response = await _client.GetAsync($"/consultations/api/Documents?consultationId={consultationId}");
			var responseString = await response.Content.ReadAsStringAsync();

	        //Assert
	        response.StatusCode.ShouldBe(HttpStatusCode.InternalServerError);
	        responseString.ShouldMatchApproved(new Func<string, string>[] { Scrubbers.ScrubErrorMessage });
		}

        [Fact]
        public async Task Get_Documents_Feed_Returns_Populated_Feed()
        {
            //Arrange (in base constructor)

            // Act
            var response = await _client.GetAsync("/consultations/api/Documents?consultationId=54");
            response.EnsureSuccessStatusCode();
            var responseString = await response.Content.ReadAsStringAsync();

            // Assert
            responseString.ShouldMatchApproved();
        }

        [Fact]
        public async Task Get_Documents_Feed_Fills_In_Blank_Document_Titles()
        {
            //Arrange (in base constructor)
            FeedToUse = Feed.ConsultationCommentsPublishedDetailNoTitle;

            // Act
            var response = await _client.GetAsync("/consultations/api/Documents?consultationId=54");
            response.EnsureSuccessStatusCode();
            var responseString = await response.Content.ReadAsStringAsync();

            // Assert
            responseString.ShouldMatchApproved();
        }

    }
}
