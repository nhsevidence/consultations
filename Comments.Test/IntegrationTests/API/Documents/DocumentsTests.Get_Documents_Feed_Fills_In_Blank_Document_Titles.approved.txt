[
    {
        "consultationId":54,
        "documentId":1,
        "convertedDocument":true,
        "supportsComments":true,
        "supportsQuestions":true,
        "title":"tophat_card.docx",
        "chapters":[
            {
                "slug":"tophatcarddocx",
                "title":"tophat_card.docx"
            }
        ],
        "href":"/guidance/GID-NG10168/documents/accessible-version",
        "sourceURI":"consultations://./consultation/54/document/1"
    },
    {
        "consultationId":54,
        "documentId":2,
        "convertedDocument":true,
        "supportsComments":true,
        "supportsQuestions":true,
        "title":"Return to work form 2017.docx",
        "chapters":[
            {
                "slug":"return-to-work-form-2017docx",
                "title":"Return to work form 2017.docx"
            }
        ],
        "href":"/guidance/GID-NG10168/documents/193",
        "sourceURI":"consultations://./consultation/54/document/2"
    },
    {
        "consultationId":54,
        "documentId":3,
        "convertedDocument":true,
        "supportsComments":true,
        "supportsQuestions":true,
        "title":"QA_checklist.docx",
        "chapters":[
            {
                "slug":"qachecklistdocx",
                "title":"QA_checklist.docx"
            }
        ],
        "href":"/guidance/GID-NG10168/documents/193-2",
        "sourceURI":"consultations://./consultation/54/document/3"
    },
    {
        "consultationId":54,
        "documentId":0,
        "convertedDocument":false,
        "supportsComments":false,
        "supportsQuestions":false,
        "title":"Addendum to draft scope",
        "chapters":null,
        "href":"/guidance/GID-NG10168/documents/addendum-to-draft-scope",
        "sourceURI":null
    },
    {
        "consultationId":54,
        "documentId":0,
        "convertedDocument":false,
        "supportsComments":false,
        "supportsQuestions":false,
        "title":"Addendum appendicies",
        "chapters":null,
        "href":"/guidance/GID-NG10168/documents/addendum-appendicies",
        "sourceURI":null
    },
    {
        "consultationId":54,
        "documentId":0,
        "convertedDocument":false,
        "supportsComments":false,
        "supportsQuestions":false,
        "title":"Draft scope",
        "chapters":null,
        "href":"/guidance/GID-NG10168/documents/draft-scope",
        "sourceURI":null
    },
    {
        "consultationId":54,
        "documentId":0,
        "convertedDocument":false,
        "supportsComments":false,
        "supportsQuestions":false,
        "title":"Database of treatment effects user guide",
        "chapters":null,
        "href":"/guidance/GID-NG10168/documents/database-of-treatment-effects-user-guide",
        "sourceURI":null
    },
    {
        "consultationId":54,
        "documentId":0,
        "convertedDocument":false,
        "supportsComments":false,
        "supportsQuestions":false,
        "title":"Database of treatment effects",
        "chapters":null,
        "href":"/guidance/GID-NG10168/documents/database-of-treatment-effects",
        "sourceURI":null
    },
    {
        "consultationId":54,
        "documentId":0,
        "convertedDocument":false,
        "supportsComments":false,
        "supportsQuestions":false,
        "title":"DOI",
        "chapters":null,
        "href":"/guidance/GID-NG10168/documents/97",
        "sourceURI":null
    }
]